src/usdb_syncer/gui/forms/ReportDialog.ui

```diff
@@ -0,0 +1,318 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
I think the source selector should be available across all report kinds, and Ok should generate the report. You'd still have the tab widget to choose the report kind and specific settings. You'd also need a Cancel button.

At this point you can also add a check if the selector matches any songs at all.